### ISSUE TITLE
Multiple filters on API not working correctly

### ISSUE THREAD

**Issue by Melf11** [Icinga/icinga2#8701]
I'm developing an own Angular Platform to visualize the data of Icinga2. I'm getting all the data via the Icinga API, but for features like infinite scrolling and for performance it is useful to use multiple filters on a single API Call.
In some cases it works, but mostly i'm getting only the last filter i added as query param.

- As example i need to filter services by hostname and the state of the service:
`/objects/services?filter=service.host_name=="<HOST-NAME>"&service.state=="3"`

i tried a lot of possibilities to combine these two filters but it is just not possible, in most common tries it is just recognized the last filter.

Another example with two filters works correctly but the same filter in different versions are not working:
`/objects/hosts?filter=host.state==1&<HOST-GROUP-NAME> in host.groups` <- is OK

- i'm expecting this filter will got me the same results but i'm getting a 404:
`/objects/hosts?filter=host.state==1&filter=host.groups=="<HOST-GROUP-NAME>"` <- 404

- and this one is just recognizing the last filter but ignoring the host-groups filter:
`/objects/hosts?filter=match("<HOST-GROUP-NAME>*",host.groups)&filter=host.state==1` <- just filters host.state

## Environment

- I don't have access to the icinga Server at the moment but it should be Version: r2.12.1-1
- Icinga Web is not running/in use
- Developing a Web Frontend with Angular 11
- Working on MacOS 11.2.1

**Comment by Al2Klimov** [comment]
Hello @Melf11 and thank you for reporting!

Have you tried combining your filters with `&&`/`||` to a single filter?

Best,
AK

**Comment by Melf11** [comment]
> Hello @Melf11 and thank you for reporting!
> 
> Have you tried combining your filters with `&&`/`||` to a single filter?
> 
> Best,
> AK

yes i did, but nothing changed. If a multiple filter with a double '&' works, than a single one will work as well.

**Comment by julianbrost** [comment]
Are you URL-encoding the `&&` properly? Otherwise it probably gets treated as a separator for multiple parameters. So please try to use `%26%26` instead of `&&`.
